codebuild/bin/install-shared-deps-awslc.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script is similar to |install-shared-deps.sh| except below differences:
+# 1. OpenSSL is replaced with awslc.
+# 2. LibCurl version is updated to 7.74.
+# After this script, awslc(static and shared) and curl will be installed under /deps/install.
+
+set -euxo pipefail
+
+export AWSLC_SRC_DIR=/tmp/awslc
+export INSTALL_DIR=/deps/install
+export LD_LIBRARY_PATH=${INSTALL_DIR}
+export NUM_CPU_THREADS=$(nproc)
+
+function download_awslc() {
+    AWSLC_GIT_URL='https://github.com/awslabs/aws-lc.git'
+    AWSLC_GIT_REF='main'
```

Comment:
We should pin a particular AWS-LC version as we do for the OpenSSL Dockerfiles. I tried changing this to the latest relevant tag I saw (AWS-LC-FIPS-1.0.2), but the Docker image failed to build because (I assume) that tag is a bit out of date. Can you tag the working commit on `main` and pin that tag here? Then I can build the image and set up a CI build that runs within it.

Reply:
Sorry for the confusion, the `AWS-LC-FIPS-1.0.2` branch points to a FIPS version release, which points to a branch of an older commit of the repo that's still in the process of getting FIPS validated.
I've set the docker image tag to `v1.1.0`, which points to a newer commit should work. Just so you know, we'll also be setting up an encryption-sdk CI in AWS-LC, so we can verify that all future versions of AWS-LC will build with ESDK :)